### DISCUSSION TITLE
Set "Stock Ownership" to existing customers, and allow delivery here

### DISCUSSION THREAD

**Discussion by ctft1** [inventree/InvenTree#6122] [Ideas]
Hi,

I'm currently evaluating Inventree for our business usage, and I have a question regarding our inventory / sale / consignment use case.

Here is a classic scenario in our activity :
 - The customer buy some products
 - He can choose between 1. immediate delivery (classic and easy), or 2. ask us to store the products for him
 - When option 2, we can keep the products from a few months... to many years
 - Then the customer can choose some products he bought from his own stock (stored in our warehouse) and deliver them to himself

Of course those "consigned" products should be:
 - Not included in our own stock (they have been sold, and don't belong to us anymore)
 - Should be tracked to be able to easily find them (product location, move, ..)
 - Should be counted with some criteria (product type), as we charge monthly / annual fees for the storage
 - Have some reports on demand (full inventory list for a customer)

This should be possible for hundreds of customers.

I've seen the "Stock Ownership" option, but I'm not sure it's the right approach as it requires a login / password to Inventree if I'm right, and our customers should not be able to change product locations, or do any other change... I would say the best option would be to allow the "Stock Ownership" to be set to an existing customer (from the "Sell > Customers" tab

Then we should be able, on the sale order, to deliver to a specific warehouse location (owned by the customer) instead of a traditional address.

Would it be easy to add such a feature?

Thank you!

NB: I've also seen the "Assign to customer" feature but I'm not sure it would work as intended (to me it more looks like a feature to temporarily assign the stock to avoid selling it twice, until shipping)

regards,

**Comment by SchrodingersGat** [comment]
Inventree is not designed to have customers modifying (or even viewing) data. However either he available data it should be reasonably straightforward to implement a custom plugin to perform the functions you need. You would of course need to consider the associated security and privacy implications of exposing the data

**Comment by ctft1** [comment]
Sorry I may not have explained so well (I'm not English native speaking)
It's fine if customers cannot modify / view the data (we can produce periodic reports for them)

The principal need is more about:
- being able to "ship" a customer order to an inventory location managed by us in our warehouse (this explains the idea of using ownership to distinguish the stock, as we cannot sell it anymore)
- keeping it possible to manage this stock even if it does not belong to us anymore (mainly move products between sub-location)

**Reply by SchrodingersGat** [reply]
Ok so more about managing item lifecycle after it has been sold or shipped to a customer? This should be possible without too much hassle. In fact most of this is possible already. Where in particular does the concept not work for you with the existing setup?

**Reply by ctft1** [reply]
Sorry I was pretty busy during the last weeks, and had to postpone my trials.
I only tested Inventree through the online demo, so maybe I had not a full access to possible options. I will try with an installed version.

If I remember well, the main problem would be assigning the ownership during the order process: either the stock can be "reserved" (but it would remain in our stock, which is not what I'm looking for) either the order has to be shipped, and in this case the stock is just "deleted"

What I would need is a way to close the order by assigning the ownership to the buyer (without shipping)

**Reply by SchrodingersGat** [reply]
Stock is not "deleted" when you ship to a customer - it is assigned to the customer and remains in the system, but is not available for further use

**Reply by ctft1** [reply]
Yes that's exactly what I've seen (I spent some time on further testing with the online demos yesterday). 
My need would be to have a kind of ownership transfer without shipping (but also further than just assigning stock)

I still need to test further, right now I'm facing the following limitations:

- An order cannot be completed if not shipped, and if shipped I loose control over the stock (even if I can see it as shipped) for further management (as the stock is still managed by us even if sold)
- the "stock ownership" can only be set to a user (user = with an Inventree login if I'm right) and not a customer (internal data). Is there a way to link a user to a customer?

**Reply by ctft1** [reply]
Also maybe I'm not understanding well the purpose of "stock ownership"... Maybe it's only intended to allow internal users (from the company) to manage the stock...  In my need, the stock ownership would be closer to a "fulfilled by our company warehouse" (kind of dropship) management: once a client owns the stock, it is still managed by us, but does not belong to our company anymore

**Comment by nmcc1212** [comment]
Did you find a solution to this, have a similar situation
